pkg/serializer/internal/metrics/sketch_series_list.go

```diff
@@ -112,222 +256,168 @@ func (sl SketchSeriesList) MarshalSplitCompress(bufferContext *marshaler.BufferC
+
+				err = ps.Sint32Packed(dogsketchK, k)
+				if err != nil {
+					return err
+				}
 
-			err = ss.Tags.ForEachErr(func(tag string) error {
-				return ps.String(sketchTags, tag)
+				err = ps.Uint32Packed(dogsketchN, n)
+				if err != nil {
+					return err
+				}
+
+				return nil
 			})
 			if err != nil {
 				return err
 			}
-
-			for _, p := range ss.Points {
-				err = ps.Embedded(sketchDogsketches, func(ps *molecule.ProtoStream) error {
-					b := p.Sketch.Basic
-					k, n := p.Sketch.Cols()
-
-					err = ps.Int64(dogsketchTs, p.Ts)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Int64(dogsketchCnt, b.Cnt)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Double(dogsketchMin, b.Min)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Double(dogsketchMax, b.Max)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Double(dogsketchAvg, b.Avg)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Double(dogsketchSum, b.Sum)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Sint32Packed(dogsketchK, k)
-					if err != nil {
-						return err
-					}
-
-					err = ps.Uint32Packed(dogsketchN, n)
+		}
+		err = ps.Embedded(sketchMetadata, func(ps *molecule.ProtoStream) error {
+			return ps.Embedded(sketchMetadataOrigin, func(ps *molecule.ProtoStream) error {
+				if ss.NoIndex {
+					err = ps.Int32(sketchMetadataOriginMetricType, metryTypeNotIndexed)
 					if err != nil {
 						return err
 					}
-
-					return nil
-				})
+				}
+				err = ps.Int32(sketchMetadataOriginOriginProduct, serieMetadataOriginOriginProductAgentType)
 				if err != nil {
 					return err
 				}
-			}
-			err = ps.Embedded(sketchMetadata, func(ps *molecule.ProtoStream) error {
-				return ps.Embedded(sketchMetadataOrigin, func(ps *molecule.ProtoStream) error {
-					if ss.NoIndex {
-						err = ps.Int32(sketchMetadataOriginMetricType, metryTypeNotIndexed)
-						if err != nil {
-							return err
-						}
-					}
-					err = ps.Int32(sketchMetadataOriginOriginProduct, serieMetadataOriginOriginProductAgentType)
-					if err != nil {
-						return err
-					}
-					err = ps.Int32(sketchMetadataOriginOriginCategory, metricSourceToOriginCategory(ss.Source))
-					if err != nil {
-						return err
-					}
-					return ps.Int32(sketchMetadataOriginOriginService, metricSourceToOriginService(ss.Source))
-				})
+				err = ps.Int32(sketchMetadataOriginOriginCategory, metricSourceToOriginCategory(ss.Source))
+				if err != nil {
+					return err
+				}
+				return ps.Int32(sketchMetadataOriginOriginService, metricSourceToOriginService(ss.Source))
 			})
-			if err != nil {
-				return err
-			}
-
-			return nil
 		})
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		// Compress the protobuf metadata and the marshaled sketch
-		err = compressor.AddItem(buf.Bytes())
-		switch err {
-		case stream.ErrPayloadFull:
-			expvarsPayloadFull.Add(1)
-			tlmPayloadFull.Inc()
+		return nil
```

Comment:
L351-354 can eventually be simplified by:

```diff
- if err != nil {
-     return err
- }
- return nil
+ return err
```

Reply:
Good idea. It looks like most of these were already here, but might as well simplify while we're here.